plugins/sinks/columbus/sink.go

```diff
@@ -41,57 +41,80 @@ func (s *Sink) Sink(ctx context.Context, configMap map[string]interface{}, in <-
 	}
 
 	for data := range in {
-		dataMapper := s.getDataMapper(data)
-		data = dataMapper(data)
+		data, err = s.mapData(data)
+		if err != nil {
+			return errors.Wrap(err, "error mapping data")
+		}
 		if err = s.send(data); err != nil {
-			return
+			return errors.Wrap(err, "error sending data")
 		}
 	}
 
 	return
 }
 
-func (s *Sink) getDataMapper(data interface{}) func(interface{}) interface{} {
-	if s.cachedDataMapper == nil {
-		s.cachedDataMapper = s.buildDataMapper(data)
+func (s *Sink) mapData(data interface{}) (interface{}, error) {
```

Comment:
@ravisuhag We need this function to map fields (e.g. `resource.urn` to `urn`) so Columbus will accept the payload.

I gave myself a time to find a better approach but I could not orchestrate a better way to map these **nested fields** before sending to Columbus without making it overly complicated. I tried a few efforts using `reflect` library and it made it super complicated and without success.

So now I'm using the most easier approach just to get it working.

Reply:
Works for now. Anyway, we are likely to change Columbus contract and there might be some changes in models as well.